pymc/step_methods/external/base.py

```diff
@@ -0,0 +1,124 @@
+#   Copyright 2024 - present The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from abc import ABC, abstractmethod
+
+from arviz import InferenceData
+
+from pymc.step_methods.compound import BlockedStep, Competence
+
+
+class ExternalSampler(BlockedStep, ABC):
+    """Base class for external samplers.
+
+    External samplers manage their own MCMC loop rather than using PyMC's.
+    These samplers (like NutPie, BlackJax, etc.) are designed to run
+    their own efficient loop inside their implementation.
+
+    Attributes
+    ----------
+    is_external : bool
+        Flag indicating that this is an external sampler that needs
+        special handling in PyMC's sampling loops.
+    """
+
+    is_external = True
```

Comment:
The base class already tells us this is an ExternalSampler, no need for `is_external`?